app/assets/javascripts/components/MessageList.js.jsx.coffee

```diff
@@ -0,0 +1,17 @@
+@MessageList = React.createClass
+  propTypes:
+    messages: React.PropTypes.array.isRequired
+
+  componentDidUpdate: ->
+    App.scrollBottom()
+
+  render: ->
+    list = @props.messages.map (message, i) ->
+      message.id = "new-#{i}" if typeof message.id == 'undefined'
```

Comment:
`key` in loops of React components are described here:
* https://facebook.github.io/react/docs/reconciliation.html#keys
* https://facebook.github.io/react/docs/multiple-components.html#dynamic-children

I'm not 100% sure, but it's probably not necessary for the case of new messages without an id.  However, React was giving me warnings in the console, so I added this just to be safe.  To me, it makes sense to just always have a `key` on something that's in a loop.